packages/schema/src/model.ts

```diff
@@ -483,6 +485,76 @@ export class JupyterCadDoc
     this
   );
 }
+
+// TODO Change the IJupyterCadDoc interface to add an "editable" property allowing to make read-only docs?
+export class JupyterCadStepDoc extends JupyterCadDoc {
+  constructor() {
+    super();
+
+    this._source = this.ydoc.getText('source');
+
+    this._source.observeDeep(this._sourceObserver);
+  }
+
+  get version(): string {
+    return '0.1.0';
+  }
+
+  get objectsChanged(): ISignal<IJupyterCadDoc, IJcadObjectDocChange> {
+    return this._objectChanged;
+  }
+
+  get objects(): Array<IJCadObject> {
+    const source = this._source.toJSON();
+
+    console.log('Source is', this._source, source);
+
+    if (!source) {
+      return [];
+    }
+
+    return [
+      {
+        name: 'Step File', // TODO get file name?
```

Comment:
Yes, i think the file name is better

Reply:
Unfortunately the ydoc does not known anything about the name of the file it originated from, or am I missing something?

Reply:
ah indeed. So let's keep it like this.